src/ecalc/libraries/libecalc/common/libecalc/input/mappers/create_references.py

```diff
@@ -26,6 +27,31 @@ def create_references(configuration: PyYamlYamlModel, resources: Resources) -> R
         facility_inputs=facility_inputs_from_files,
         resources=resources,
     )
+
+    duplicated_fuel_names = get_duplicates(
+        [fuel_data.get(EcalcYamlKeywords.name) for fuel_data in configuration.fuel_types]
+    )
+
+    if len(duplicated_fuel_names) > 0:
+        raise ValueError(
+            "Fuel type names must be unique across installations."
+            f" Duplicated names are: {', '.join(duplicated_fuel_names)}"
+        )
+
+    fuel_types_emissions = [list(fuel_data[EcalcYamlKeywords.emissions]) for fuel_data in configuration.fuel_types]
+
+    # Check each fuel for duplicated emissions
+    duplicated_emissions = []
+    for emissions in fuel_types_emissions:
```

Comment:
instead of grabbing all failures here, I would perhaps want to just break on the first duplicate we find, to make the logic easier to grasp. Should not happen that often after all...